gtk/src/callbacks.c

```diff
@@ -577,6 +579,70 @@ check_name_template(signal_user_data_t *ud, const char *str)
     return FALSE;
 }
 
+int match(const char *string, const char *pattern)
+{
+    int status;
+    regex_t re;
+    if (regcomp(&re, pattern, REG_EXTENDED|REG_NOSUB) != 0)
+    {
+        return(0);
+    }
+    status = regexec(&re, string, (size_t) 0, NULL, 0);
+    regfree(&re);
+    if (status != 0) 
+    {
+        return(0);
+    }
+    return(1);
+}
+
+typedef struct {
+    const char *pattern;
+    const char *format;
+} datemap;
+
+static int
+parse_datestring(const char *src, struct tm *tm)
+{       
+    datemap ymdThmsZ = {"[0-9]{4}-[0-1]?[0-9]-[0-3]?[0-9]T[0-9]{2}:[0-9]{2}:[0-9]{2}Z", "%Y-%m-%dT%H:%M:%SZ"};  
+
+    datemap maps[1] = { ymdThmsZ };
+
+    for (int i = 0; i < sizeof(maps); i++)
+    {
+        if (match(src, maps[i].pattern))
+        {
+            strptime(src, maps[i].format, tm);
+            return(1);
+        }
+    }
+    return(0);
+}
+
+static char*
+get_creation_date(const char *pattern, const char *metaValue, const char *file, char **str)
+{       
+    char date[11];       
+    if (metaValue != NULL && strlen(metaValue) > 1)
+    {
+        struct tm tm;
+        if (parse_datestring(metaValue, &tm))
+        {            
+            strftime(date, 11, pattern, &tm);                                                    
+        }        
+    }
+    else
+    {
+        struct stat stbuf;
+        if (g_stat(file, &stbuf) == 0){
+            struct tm *tm;
+            tm = localtime(&(stbuf.st_mtime));                                    
+            strftime(date, 11, pattern, tm);                                                    
+        }
+    }
+    *str = date;    
```

Comment:
Your passing back a pointer to something on the stack.  Since the stack frame for this function is no longer valid after returning, the string may get stepped on before it's used.

I suggest *str = strdup(date). Then free the string where ever get_creation_date is used.